expression/builtin_time.go

```diff
@@ -2516,7 +2516,42 @@ type builtinTimeFormatSig struct {
 // eval evals a builtinTimeFormatSig.
 // See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_time-format
 func (b *builtinTimeFormatSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("TIME_FORMAT")
+	args, err := b.evalArgs(row)
```

Comment:
hi, we are currently refactoring our expression evaluation framework, plz rewrite `time_format` under the new framework. references:
1. https://zhuanlan.zhihu.com/p/27528657
2. https://zhuanlan.zhihu.com/p/27928453